src/mango/src/mango_idx_special.erl

```diff
@@ -58,10 +58,16 @@ columns(#idx{def = all_docs}) ->
     [<<"_id">>].
 
 is_usable(#idx{def = all_docs}, _Selector, []) ->
-    true;
+    {true, #{reason => []}};
 is_usable(#idx{def = all_docs} = Idx, Selector, SortFields) ->
     Fields = mango_idx_view:indexable_fields(Selector),
-    lists:member(<<"_id">>, Fields) and can_use_sort(Idx, SortFields, Selector).
+    SelectorHasRequiredFields = lists:member(<<"_id">>, Fields),
+    CanUseSort = can_use_sort(Idx, SortFields, Selector),
+    Reason =
+        [field_mismatch || not SelectorHasRequiredFields] ++
+            [sort_order_mismatch || not CanUseSort],
```

Comment:
This is bit of an odd pattern but it works well here and it's kind of neat!

Reply:
I am glad you liked it!